### ISSUE TITLE
[Bug] runtimeerror: size mismatch (got input: [11], target: [1]) crossentropyloss

### ISSUE THREAD

**Issue by kyre-99** [open-mmlab/mmclassification#1163]
### Branch

master branch (0.24 or other 0.x version)

### Describe the bug

caculate crossentropyloss does not work when the batch size is setted 8.
rather when i set 16 ,it work and does not show bug.
i do want to know why? the batchsize has requirement?
by the way,the items of my dataset are over 8 absolutely.

### Environment

python tools/train.py config/lenet5/lenet5_polsar.py

**Comment by Ezra-Yu** [comment]
Can you show your config?

**Comment by kyre-99** [comment]
# model settings
model = dict(
    type='ImageClassifier',
    backbone=dict(type='LeNet5', num_classes=4),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
    ))
# dataset settings

train_pipeline = [
    dict(type='LoadNPFromFile'),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label']),
]
test_pipeline = [
    dict(type='LoadNPFromFile'),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img']),
]
data = dict(
    samples_per_gpu=16,
    workers_per_gpu=2,
    train=dict(
        type='PolSar',
        data_prefix='data/polsar/train4', 
        classes='data/polsar/label_names4.txt',
        pipeline=train_pipeline),
    val=dict(
        type='PolSar', data_prefix='data/polsar/validation4', 
        classes='data/polsar/label_names4.txt',
        pipeline=test_pipeline),
    test=dict(
        type='PolSar', data_prefix='data/polsar/validation4', 
        classes='data/polsar/labes_nams4.txt',
        pipeline=test_pipeline))


log_config = dict(
    interval=150,
    hooks=[
        dict(type='TextLoggerHook'),
        # dict(type='TensorboardLoggerHook')
    ])

evaluation = dict(
    interval=1, metric='accuracy', metric_options={'topk': (1,3 )})
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/polsar4/'
load_from = None
resume_from = None
workflow = [('train', 1)]
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(policy='step', gamma=0.98, step=[15])
runner = dict(type='EpochBasedRunner', max_epochs=50)
checkpoint_config = dict(interval=1)

**Comment by mzr1996** [comment]
Please provide the entire error message with the traceback.
